pkgs/games/angband/default.nix

```diff
@@ -13,25 +13,53 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-XH2FUTJJaH5TqV2UD1CKKAXE4CRAb6zfg1UQ79a15k0=";
   };
 
-
-  nativeBuildInputs = [ autoreconfHook ];
-  buildInputs = [ ncurses5 ]
-  ++ lib.optionals enableSdl2 [
-    SDL2
-    SDL2_image
-    SDL2_sound
-    SDL2_mixer
-    SDL2_ttf
-  ];
-
-  configureFlags = lib.optional enableSdl2 "--enable-sdl2";
-
-  installFlags = [ "bindir=$(out)/bin" ];
-
   meta = with lib; {
     homepage = "https://angband.github.io/angband";
     description = "A single-player roguelike dungeon exploration game";
     maintainers = [ maintainers.kenran ];
     license = licenses.gpl2;
+    platforms = platforms.unix;
   };
+
+  # the toolchain on darwin can't build universal binaries
+  arch = if stdenv.isAarch64
+         then "arm64"
+         else "x86_64";
+in
+stdenv.mkDerivation {
+  inherit pname version src meta;
+
+  nativeBuildInputs = lib.optionals stdenv.isLinux [ autoreconfHook ];
+  buildInputs = lib.optionals stdenv.isLinux [ ncurses5 ]
+    ++ lib.optionals enableSdl2 [
+      SDL2
+      SDL2_image
+      SDL2_sound
+      SDL2_mixer
+      SDL2_ttf
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk_11_0.frameworks.Cocoa
+      darwin.stubs.setfile
+    ];
+
+  buildPhase = if stdenv.isDarwin
+    then ''
+      pushd src
```

Comment:
You need to add `runHook preBuild` and `runHook postBuild` when overriding `buildPhase`: https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#phases-sec-stdenv-phases